runners/direct-java/src/main/java/org/apache/beam/runners/direct/TestStreamEvaluatorFactory.java

```diff
@@ -130,13 +157,10 @@ public TransformResult finishBundle() throws Exception {
             .advance(((ProcessingTimeEvent<T>) event).getProcessingTimeAdvance());
       }
       index++;
-      checkState(inUse.compareAndSet(true, false),
-          "The InUse flag of a %s was changed while the source evaluator was executing. "
-              + "%s cannot be split or evaluated in parallel.",
-          TestStream.class.getSimpleName(),
-          TestStream.class.getSimpleName());
+      cache.release(application, this);
```

Comment:
`finally`'d

This doesn't stop an exception from outside the evaluator itself from making this get stuck, but we currently are unable to recover from those exceptions anyway.